### DISCUSSION TITLE
Can I make the code shorter ?

### DISCUSSION THREAD

**Discussion by eitan-rosenberg** [classgraph/classgraph#572] [Q&A] [answered]
I have this piece of code:

``` 
	try (final ScanResult scanResult = new ClassGraph().acceptClasses(name).scan()) {

			final ClassInfo classInfo = scanResult.getClassInfo(name);

			final Resource resource = classInfo.getResource();

			final File file = resource.getClasspathElementFile();

			System.out.println(file);

		}
```
		
**It is working just fine, can I make the code shorter ?**

Thanks for any response...

Regards.

**Comment by lukehutch** [comment, marked as answer]
Yes, it is totally fine to chain these calls -- the API is designed to do this sort of thing.

```java
try (final ScanResult scanResult = new ClassGraph().acceptClasses(name).scan()) {
    final File file = scanResult.getClassInfo(name).getResource().getClasspathElementFile();
    System.out.println(file);
}
```

The only thing you can't really avoid is using the try-with-resources pattern, since there may be resources that need to be closed.

**Comment by lukehutch** [comment]
If you're asking if there's a shorter way to get the classpath element that a certain class is contained within, no, there is no way currently to get that. I could probably provide a `Map<String, List<Resource>> ScanResult.getAllClasspathResources()` mapping from classpath entry path to all resources in that classpath entry, but I don't know how many people would actually need that.
